File: src/compiler/emitter.py
class Emitter:
    def __init__(self, file):
        self.file = file
        self.header = ""
        self.code = ""

    def emit(self, content):
        self.code += content

    def emit_line(self, content):
        self.code += f"{content}\n"

    def emit_header(self, content):
        self.header += f"{content}\n" 

    def write_file(self):
        try:
            # creates file if it doesn't exist and opens it in (over)write mode
            f = open(f"{self.file}.c", "w+")
            f.write(f"{self.header}\n{self.code}")
        except:
            print("ERROR: something went wrong during compilation.")

__all__ = [
    "Emitter"
]

if __name__ == "__main__":
    import os
    raise Exception(f"`{os.path.basename(__file__)} can\'t be ran as `{__name__}`")

File: src/compiler/emitter.js
const fs = require("fs")
const process = require("process")

class Emitter {
    constructor(file) {
        this.file = file
        this.header = ""
        this.code = ""
    }

    emit(content) {
        this.code += content
    }

    emit_line(content) {
        this.code += `${content}\n`
    }

    emit_header(content) {
        this.header += content
    }

    emit_header_line(content) {
        this.header += `${content}\n`
    }

    write_file() {
        try {
            fs.writeFileSync(`${this.file}.c`, `${this.header}\n${this.code}`)
        } catch(error) {
            console.error("ERROR: something went wrong during compilation.", err)
            console.error(error)
            process.exit(1)
        }
    }
}

module.exports = {
    Emitter
}

if (require.main === module) {
    const path = require("path")

    console.error(`ERROR: \`${path.parse(__filename).name}\` file can\'t be ran as main.`)
    process.exit(1)
}